expertise/models/multifacet_recommender/specter.py

```diff
@@ -211,7 +212,6 @@ def set_archives_dataset(self, archives_dataset):
                                 "authors": [profile_id],
                                 "mdate": pub_mdate
                             }
-                        self._remove_keys_from_cache(publication["id"])
```

Comment:
so this is the call that is taking time? `tensorset` is fast enough?

Reply:
Yes, tensorset is fast but the `_remove_keys_from_cache` function scans through all the keys to find matches which is quite slow

Reply:
How much is slow? I'm still confused as to why redis is slow here. Are there too many keys in the database?

I'm not sure what the `self._remove_keys_from_cache` method does, but its performance also depends on whether [`SCAN` or `KEYS`](https://stackoverflow.com/questions/32603964/scan-vs-keys-performance-in-redis) is used.